presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryStreamReader.java

```diff
@@ -160,17 +158,11 @@ public void readBatch(Object vector)
             inDictionaryStream.getSetBits(nextBatchSize, inDictionary, isNullVector);
         }
 
-        for (int i = 0; i < nextBatchSize; i++) {
-            if (isNullVector[i]) {
-                sliceVector.vector[i] = null;
-            }
-            else if (inDictionary[i]) {
-                sliceVector.vector[i] = dictionary[dataVector[i]];
-            }
-            else {
-                sliceVector.vector[i] = rowGroupDictionary[dataVector[i]];
-            }
-        }
+        int[] ids = Arrays.copyOfRange(dataVector, 0, nextBatchSize);
```

Comment:
We know do not know the batchSize when we create the readers, we compute it just before we start reading.